docs/rules/init-declarations.md

```diff
@@ -0,0 +1,87 @@
+# Enforce/Disallow Variable Initializations (init-declarations)
+
+In JavaScript, variables can be assigned during declaration, or at any point afterwards using an assignment statement. For example, in the following code, `foo` is initialized during declaration, while `bar` is initialized later.
+
+```js
+var foo = 1;
+var bar;
+
+if (foo) {
+    bar = 1;
+} else {
+    bar = 2;
+}
+```
+
+## Rule Details
+
+This rule is aimed at enforcing or eliminating variable initializations during declaration. For example, in the following code, `foo` is initialized during declaration, while `bar` is not.
+
+```js
+var foo = 1;
+var bar;
+
+bar = 2;
+```
+
+This rule aims to bring consistency to variable initializations and declarations.
+
+### Options
+
+This rule is configured by passing in the string `"always"` (the default) to enforce initialization at declaration, or `"never"` to disallow initialization during declaration. This rule applies to `var`, `let`, and `const` variables, however `"never"` is ignored for `const` variables. Because `const`s cannot be reassigned, it does not make sense to prevent them from being initialized.
+
+You can configure the rule as follows:
+
+```javascript
+{
+    // (default) All variables must be initialized at declaration
+    "init-declarations": [2, "always"],
+
+    // Variables must not be initialized at declaration
+    "init-declarations": [2, "never"]
+}
+```
+
+When configured with `"always"` (the default), the following patterns are considered warnings:
+
+```js
+function foo() {
+    var bar;
+    let baz;
+    const qux;
+}
+```
+
+The following patterns are not considered warnings with `"always"`.
+
+```js
+function foo() {
+    var bar = 1;
+    let baz = 2;
+    const qux = 3;
+}
+```
+
+When configured with `"never"`, the following patterns are considered warnings.
+
+```js
+function foo() {
+    var bar = 1;
+    let baz = 2;
+}
+```
+
+The following patterns are not considered warnings with `"never"`. Note that `const` variables are ignored with `"never"`.
+
+```js
+function foo() {
+    var bar;
+    let baz;
+    const buzz = 1;
+    const qux;
```

Comment:
Yea, I ran into that during testing. I thought it might just be an issue with the parser, since Chrome allows it. Checked with the spec, and it is indeed an error. I have updated the documentation.